doc/manual/rule_uid_schema.md

```diff
@@ -0,0 +1,189 @@
+<!---
+This Source Code Form is subject to the terms of the Mozilla
+Public License, v. 2.0. If a copy of the MPL was not distributed
+with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
+-->
+
+# Rule UID Schema
+
+## Introduction
+
+The document describes the composition of the unique identifier (UID) to be
+associated with quality checker rules.
+
+The document describes composition of rule UIDs, declares some of the concepts
+to be used for ASAM rules, and provides an example for an UID for a rule of the
+ASAM OpenDRIVE standard.
+
+## Rule UID Concepts
+
+The UID is a string which encapsulate a sequence of **concepts** that allow to
+identify immediately a rule across the different domains. The concept are
+ordered and separated via the separation character **`:`**.
+
+The concepts for the rule UID are:
+
+ * **Emanating Entity**: a domain name for the entity (organization or company)
+   that declares the rule UID
+ * **Standard**: a short string that represents the standard or the
+   domain to which the rule is applied
+ * **Definition Setting**: the version of the standard or the domain to which
+   the rule appears or is applied **for the first time**
+ * **Rule Full Name**: the full name of the rule, as dot separated, snake lower
+   case string. The full name of a rule is composed by the **rule set**, a
+   categorization for the rule, and the rule **name**, a unique string inside
+   the categorization. The rule set can be nested (meaning that can be defined
+   as an arbitrary sequence of dot separated names, while the name is the snake
+   case string after the last dot of the full name)
+ * Optional **Version**: a counter value representing the current version of the
+   rule content. If content of the rule changes without changing the underlying
+   logic for validation, this counter increases. If the change in the rule
+   forces a modification in the validation logic, the change shall be treated as
+   a new rule.
```

Comment:
I don't know if it's necessary, but should we define a default interpretation if no version is specified? E.g. "no version specified, will be interpreted as version 1". I currently don't know the specified handling for rule versions. Is it desired that all rules in all versions are checked when the checker is executed or only the latest ones? For the latter interpretation, a standard interpretation for no version information would be necessary.

Reply:
Later I found the following passage: 

> A query engine implementation shall return the most recent rule version available

so a standard interpretation is necessary, I think.

Reply:
The rule version is used only to define a change in wording in the rule that does not mandate a change in the logic for checking the rule (e.g., rephrasing), or change in its metadata description. I suppose change in rule version will almost never happen.

 * In terms of _definition_ of the rule (e.g., when the uid is firstly defined), I think we should consider version 1 as default, if no other version is available (i.e., the rule version is mandatory as explicit from rule version 2)
 * In terms of _searching_ the rule, the query implementation shall always consider the latest version available, and not 1 by default.
 
Still, **I think I should explain this better in the docs**, but before doing so I just want to confirm if we agree on the distinction between _definition_ and _search_.

Reply:
Yes, I agree with you on the distinction and also on being mandatory from rule version 2.

And regarding the search, I just wanted to clarify that we take the rule version "" as "1". So given the following rules:
`asam.net:xodr:1.6.0:road.planview.geometry.ref_line_exists:`
and
`asam.net:xodr:1.6.0:road.planview.geometry.ref_line_exists:2`

A search for the latest rule version would result in:
`asam.net:xodr:1.6.0:road.planview.geometry.ref_line_exists:2`

The term "default" I used was probably confusing. I meant the interpretation of version "1" for no particular value: "".

Reply:
Isn't versioning the job of git?
If the UID containing a version number somehow translates to a concrete file name (e.g. xodr_1.6.0_road.planview.geometry.ref_line_exists_**1**.py vs.
xodr_1.6.0_road.planview.geometry.ref_line_exists_**2**.py), 
git would no longer be able to maintain the history of this rule.

Reply:
I don't think the rule name 
`asam.net:xodr:1.6.0:road.planview.geometry.ref_line_exists:` 
would directly infer to a filename 
`xodr_1.6.0_road.planview.geometry.ref_line_exists_1.py`

From my understanding there could be a script called `ref_line_exists.py`. This declares, for exmaple, via a variable it checks the rule `asam.net:xodr:1.6.0:road.planview.geometry.ref_line_exists:` (or multiple rules `asam.net:xodr:1.6.0:road.planview.geometry.ref_line_exists:`, `asam.net:xodr:1.6.0:road.planview.geometry.ref_line_exists:2`).

In order that the checker framework can determine which module/script checks the latest version of a rule it would query for `asam.net:xodr:1.6.0:road.planview.geometry.ref_line_exists:`. This should lead to the script/module implementing: `asam.net:xodr:1.6.0:road.planview.geometry.ref_line_exists:2`.
I just wanted to clarify that if no version is specified `*:` this will be treated as `:1` during query time and as definition. You should know that if someone defines a version `:2` of a rule, the version `:` is not handled.